cedar-lean/Cedar/Thm/Authorization/Evaluator.lean

```diff
@@ -143,29 +143,29 @@ theorem and_produces_bool_or_error {e₁ e₂ : Expr} {request : Request} {entit
 := by
   cases h : evaluate (Expr.and e₁ e₂) request entities <;> simp
   case ok val =>
-    cases val <;> simp [evaluate, Result.as, Coe.coe, Value.asBool, Lean.Internal.coeM, pure, Except.pure] at h <;>
+    cases val <;> simp only [evaluate._eq_4, Result.as, Coe.coe, Value.asBool, Bool.not_eq_true',
```

Comment:
We should not use internally generated names like `evaluate._eq_4` in `simp only`.  They are flaky.  

https://github.com/leanprover/lean4/issues/3547

I manually change all of them to just the externally visible names when they are suggested, in this case, `evaluate`.